### PR TITLE
Adding api documentation for acl validation endpoint

### PR DESCRIPTION
**Pull request by AadityaChaudhary**
Includes example POST body for input and example outputs for passing and failing tests.

### CODE REVIEW

File: api.md [hunk 1]
```diff
@@ -18,6 +18,7 @@ Currently based on {some authentication method}. Visit the [admin panel](https:/
     - [GET tailnet ACL](#tailnet-acl-get)
     - [POST tailnet ACL](#tailnet-acl-post): set ACL for a tailnet
     - [POST tailnet ACL preview](#tailnet-acl-preview-post): preview rule matches on an ACL for a resource
+	- [POST tailnet ACL validation](#tailnet-acl-validation): run tests against the existing ACL.
```
Comment (soniaappasamy) [inline api.md:21]: I think you have an accidental extra tab here?

Comment (soniaappasamy) [reply api.md:21]: Also no period after ACL, just to keep things consistent with the items above

Comment (soniaappasamy) [reply api.md:21]: maybe "run validation tests against the tailnet's existing ACL"

File: api.md [hunk 2]
```diff
@@ -363,42 +364,25 @@ Etag: "e0b2816b418b3f266309d94426ac7668ab3c1fa87798785bf82f1085cc2f6d9c"
 }
 ```
 
-<a name=tailnet-acl-post></a>
```
Comment (soniaappasamy) [inline api.md:366]: Ah so I think you want to keep this section around - It looks like there's a `[POST tailnet ACL]` up above that is meant to link to this portion.

Comment (soniaappasamy) [reply api.md:366]: The new `tailnet-acl-preview-post` section looks good, I think we just want to make sure we keep the `tailnet-acl-post` one around too.

File: api.md [hunk 3]
```diff
@@ -363,42 +364,25 @@ Etag: "e0b2816b418b3f266309d94426ac7668ab3c1fa87798785bf82f1085cc2f6d9c"
...
-#### `POST /api/v2/tailnet/:tailnet/acl` - set ACL for a tailnet
+<a name=tailnet-acl-preview-post></a>
 
-Sets the ACL for the given domain.
-HuJSON and JSON are both accepted inputs.
-An `If-Match` header can be set to avoid missed updates.
+#### `POST /api/v2/tailnet/:tailnet/acl/preview` - preview rule matches on an ACL for a resource
 
-Returns the updated ACL in JSON or HuJSON according to the `Accept` header on success. Otherwise, errors are returned for incorrectly defined ACLs, ACLs with failing tests on attempted updates, and mismatched `If-Match` header and ETag.
+Determines what rules match for a user on an ACL without saving the ACL to the server.
```
Comment (soniaappasamy) [inline api.md:371]: nit: `Determines what rules match for a user on the provided ACL. This endpoint is just for previewing an ACL, and does not write anything back to the tailnet.`

File: api.md [hunk 4]
```diff
@@ -363,42 +364,25 @@ Etag: "e0b2816b418b3f266309d94426ac7668ab3c1fa87798785bf82f1085cc2f6d9c"
...
-An ACL policy may contain the following top-level properties:
-
-* `Groups` - Static groups of users which can be used for ACL rules.
-* `Hosts` - Hostname aliases to use in place of IP addresses or subnets.
-* `ACLs` - Access control lists.
-* `TagOwners` - Defines who is allowed to use which tags.
-* `Tests` - Run on ACL updates to check correct functionality of defined ACLs.
-
-See https://tailscale.com/kb/1018/acls for more information on those properties.
+ACL JSON or HuJSON (see https://tailscale.com/kb/1018/acls)
```
Comment (soniaappasamy) [inline api.md:379]: I'd make this:

```
The POST body should be a JSON or [HuJSON](https://github.com/tailscale/hujson#hujson---human-json) formatted JSON object.
See https://tailscale.com/kb/1018/acls for information on available properties.
```

File: api.md [hunk 5]
```diff
@@ -424,41 +408,50 @@ curl 'https://api.tailscale.com/api/v2/tailnet/example.com/acl' \
 
 Response:
 ```
-// Example/default ACLs for unrestricted connections.
+{"matches":[{"users":["*"],"ports":["*:*"],"lineNumber":19}],"user":"user1@example.com"}
+```
+
+<a name=tailnet-acl-validation></a>
+
+#### `POST /api/v2/tailnet/:tailnet/acl/validate` - run tests against the ACL
```
Comment (soniaappasamy) [inline api.md:416]: nit: `run validation tests against the tailnet's active ACL`

File: api.md [hunk 6]
```diff
@@ -424,41 +408,50 @@ curl 'https://api.tailscale.com/api/v2/tailnet/example.com/acl' \
...
+
+Runs the provided ACLTests against the existing ACL without modifying the ACL.
```
Comment (soniaappasamy) [inline api.md:418]:
```
Runs the provided ACL tests against the tailnet's existing ACL. This endpoint does not modify the ACL in any way.
```

**Review by soniaappasamy — COMMENTED** [review]
Looking good! Mostly just left some comments on how to rephrase some stuff. Also need to make sure you don't delete the old `tailnet-acl-post` section!

**Review by soniaappasamy — APPROVED** [review]

**Comment by bradfitz** [comment]
@soniaappasamy, if this is live in production, want to merge it?

**Comment by soniaappasamy** [comment]
Yeah will do!

**Pull request merged**